AllReadyApp/Web-App/AllReady/Areas/Admin/Controllers/EventManagerInviteController.cs

```diff
@@ -0,0 +1,114 @@
+﻿using AllReady.Areas.Admin.Features.EventManagerInvites;
+using AllReady.Areas.Admin.ViewModels.ManagerInvite;
+using AllReady.Features.Events;
+using AllReady.Models;
+using AllReady.Security;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace AllReady.Areas.Admin.Controllers
+{
+    [Area("Admin")]
+    [Authorize(nameof(UserType.OrgAdmin))]
+    public class EventManagerInviteController : Controller
+    {
+        private readonly IMediator _mediator;
+        private readonly UserManager<ApplicationUser> _userManager;
+
+        public EventManagerInviteController(IMediator mediator, UserManager<ApplicationUser> userManager)
+        {
+            _mediator = mediator;
+            _userManager = userManager;
+        }
+
+        // GET: Admin/Invite/Send/5
+        [Route("[area]/[controller]/[action]/{eventId:int}")]
+        public async Task<IActionResult> Send(int eventId)
+        {
+            var @event = await _mediator.SendAsync(new EventByEventIdQuery { EventId = eventId });
+            if (@event == null) return NotFound();
+
+            if (!User.IsOrganizationAdmin(@event.Campaign.ManagingOrganizationId))
+            {
+                return Unauthorized();
+            }
+
+            return View("Send", new EventManagerInviteViewModel
+            {
+                EventId = eventId,
+                EventName = @event.Name,
+                CampaignId = @event.CampaignId,
+                CampaignName = @event.Campaign.Name,
+            });
+        }
+
+        // POST: Admin/Invite/Send/5
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        [Route("[area]/[controller]/[action]/{eventId:int}")]
+        public async Task<IActionResult> Send(int eventId, EventManagerInviteViewModel invite)
+        {
+            if (invite == null)
+            {
+                return BadRequest();
+            }
+
+            if (ModelState.IsValid)
+            {
+                var @event = await _mediator.SendAsync(new EventByEventIdQuery { EventId = eventId });
+                if (@event == null)
+                {
+                    return BadRequest();
+                }
+
+                if (!User.IsOrganizationAdmin(@event.Campaign.ManagingOrganizationId))
+                {
+                    return Unauthorized();
+                }
+
+                if (await _mediator.SendAsync(new UserHasEventManagerInviteQuery { EventId = invite.EventId, InviteeEmail = invite.InviteeEmailAddress }))
+                {
+                    ModelState.AddModelError("InviteeEmailAddress", $"An invite already exists for {invite.InviteeEmailAddress}.");
+                    return View("Send", invite);
+                }
+
```

Comment:
@mgmccarthy We cannot make registered users searchable for org admin level since we will be potentially running the site in a shared mode. Therefore it was agreed that it would have to be invite which doesn't expose addresses. So yes, you are correct, if not registered the user will need to first signup. If the action handling the link is behind auth, then this is pretty much taken care of as the user will have to sign in, or first sign up. Then they will be able to click the invite link.

Not included yet, but I think will follow is some kind of notification when they login, if they have invites pending. That makes the flow smoother for unregistered user invites.